command/task_create.go

```diff
@@ -0,0 +1,181 @@
+	     performed if monitored services have changed.
+
+      Only 'yes' will be accepted to approve.
+
+  Enter a value: yes
+  
+  // ... output continues
+`, strings.Join(c.meta.helpOptions, "\n"))
+	return strings.TrimSpace(helpText)
+}
+
+// Synopsis is a short one-line synopsis of the command
+func (c *taskCreateCommand) Synopsis() string {
+	return "Creates a new task."
+}
+
+// Run runs the command
+func (c *taskCreateCommand) Run(args []string) int {
+	c.meta.setFlagsUsage(c.flags, args, c.Help())
+	var taskFile string
+	c.flags.StringVar(&taskFile, "task-file", "", "A file containing the hcl definition of a task")
+
+	if err := c.flags.Parse(args); err != nil {
+		return ExitCodeParseFlagsError
+	}
+
+	// Check that a task file was provided
+	if len(taskFile) == 0 {
+		c.UI.Error("Error: no task file provided")
+		help := fmt.Sprintf("For additional help try 'consul-terraform-sync %s --help'",
+			cmdTaskCreateName)
+		help = wordwrap.WrapString(help, width)
+
+		c.UI.Output(help)
+
+		return ExitCodeRequiredFlagsError
+	}
+
+	client, err := c.meta.taskLifecycleClient()
+	if err != nil {
+		c.UI.Error("Error: unable to create client")
+		msg := wordwrap.WrapString(err.Error(), uint(78))
+		c.UI.Output(msg)
+
+		return ExitCodeError
+	}
+
+	// Build a CTS config and use the config.Tasks object only
+	cfg, err := config.BuildConfig([]string{taskFile})
+	if err != nil {
+		c.UI.Error("Error: unable to read task file")
+		msg := wordwrap.WrapString(err.Error(), uint(78))
+		c.UI.Output(msg)
+
+		return ExitCodeError
+	}
+	taskConfigs := *cfg.Tasks
+
+	// Check that we have exactly 1 task in the task config return
+	l := len(taskConfigs)
+	if l > 1 {
+		c.UI.Error(fmt.Sprintf("Error: task file %s cannot contain more "+
+			"than 1 task, contains %d tasks", taskFile, l))
+		return ExitCodeError
+	}
+
+	if l == 0 {
+		c.UI.Error(fmt.Sprintf("Error: task file %s does not contain a task, "+
+			"must contain at least one task", taskFile))
+		return ExitCodeError
+	}
+
+	taskConfig := taskConfigs[0]
+	taskName := *taskConfig.Name
+
+	// First inspect the plan
+	c.UI.Info(fmt.Sprintf("Inspecting changes to resource if creating task '%s'...\n", taskName))
+	c.UI.Output("Generating plan that Consul Terraform Sync will use Terraform to execute\n")
+
+	taskReq := api.TaskRequestFromTaskRequestConfig(*taskConfig)
+	taskResp, err := client.CreateTask(context.Background(), api.RunOptionInspect, taskReq)
+
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error: unable to generate plan for '%s'", taskName))
+		err = processEOFError(client.Scheme(), err)
+		msg := wordwrap.WrapString(err.Error(), uint(78))
+		c.UI.Output(msg)
+
+		return ExitCodeError
+	}
+
+	c.UI.Output(fmt.Sprintf("Request ID: %s", taskResp.RequestId))
+	b, _ := json.MarshalIndent(taskResp.Task, "    ", "  ")
+	c.UI.Output(fmt.Sprintf("Task: %s\n", string(b)))
+	c.UI.Output(fmt.Sprintf("Plan: \n%s", *taskResp.Run.Plan))
+
+	if !*c.autoApprove {
```

Comment:
I mentioned this in the recent `auto-approve` PR and sharing the discussion here. Not necessary asking for changes to be made for this PR, but an enhancement for the `auto-approve` feature would be to skip the `RunOptionInspect` portion and just run now since auto-approve doesn't expect human interaction to begin with.

Reply:
For reference, here's the conversation around this: https://github.com/hashicorp/consul-terraform-sync/pull/576#pullrequestreview-843784086

Reply:
Great, I'll leave it as is for now, and we can tackle this enhancement along with the other locations of auto-approve